Data/Row/Records.hs

```diff
@@ -291,11 +291,28 @@ map f = unRMap . metamorph @_ @r @c @Rec @(RMap f) @Identity Proxy doNil doUncon
            => Label ℓ -> Identity τ -> RMap f ('R ρ) -> RMap f ('R (ℓ :-> τ ': ρ))
     doCons l (Identity v) (RMap r) = RMap (unsafeInjectFront l (f v) r)
 
+newtype RFMap (g :: * -> *) (ϕ :: Row (* -> *)) (ρ :: Row *) = RFMap { unRFMap :: Rec (Ap ϕ (Map g ρ)) }
+newtype RecAp (ϕ :: Row (* -> *)) (ρ :: Row *) = RecAp (Rec (Ap ϕ ρ))
+newtype App (f :: * -> *) (a :: *) = App (f a)
```

Comment:
What is the purpose of `App`?

Reply:
The `metamorph` function has 2 main parts: it unfolds a row and then it folds it back up.  During the unfold, it produces a value for each type in the row, and this value is wrapped up in an `h`.

For `biMetamorph`, the kind of `h` is `k1 -> k2 -> *`, where `k1` is the kind of the elements of the first row, and `k2` is the kind of the elements of the second row.  For `mapF`, this means we need a type that has kind `(k -> *) -> k -> *`, where the first argument is the `Functor`-ish part of the first row and the second is the type in the second row.  Therefore, I defined `App` (which acts as type-level application).  Note that it's being used as `h` in the `biMetamorph` call in `mapF`.

I was really hoping that something like `App` already existed in some form, but I couldn't find anything.  If you have an alternative, I'd be happy to use it.

Reply:
I do find it annoying that we have to write all these newtypes to do type-level operations.  But hey, we're not leaking them in the export list, and they don't take any run time, so it's not _terrible_.